### PR TITLE
Update zcash to new V5 API; add tests

### PR DESCRIPTION
**Pull request by conradoplg**
Expose the new V5 `zcash_script` functions, which implement ZIP-244 changes, see https://github.com/zcash/zcash/pull/5607

This points to that PR (it must be updated after it's merged, so it's a draft PR for now) and add tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,9 @@
 
 Rust bindings to the ECC's `zcash_script` c++ library.
 
+`zcash_script` links to `librustzcash`, which is written in Rust. Therefore,
+when updating `zcash_script`, we need to make sure that
```
Comment (teor2345) [inline README.md:17]: Incomplete docs?

**Review by teor2345 — DISMISSED** [review]
Looks good.

There's a sentence in the docs that got cut off, but it's not a blocker.

File: README.md (hunk 1 above)
Comment (conradoplg) [reply README.md]: Thanks for catching that! I fixed this in 579c0c0a2 (I force-pushed this branch because the history became a mess)

**Comment by conradoplg** [comment]
@teor2345 this is ready again - I fixed the comment. (I tried to add a V5 test but realized that particular test vector wasn't supposed to work)

**Comment by conradoplg** [comment]
Ah, forgot that we need to wait for the `zcash` PR to merge

**Review by teor2345 — APPROVED** [review]
Thanks for the tests and docs, looks great!

File: .gitmodules [hunk 2]
```diff
@@ -1,3 +1,4 @@
 [submodule "depend/zcash"]
 	path = depend/zcash
 	url = https://github.com/ZcashFoundation/zcashd.git
+	branch = nu5-consensus
```
[the remark this answers is not in the document]
Comment (conradoplg) [reply .gitmodules]: Done in ceca5845a0bf

**Review by upbqdn — APPROVED** [review]

**Pull request merged**